### ISSUE TITLE
Notification Fails To Dismiss (Sometimes)

### ISSUE THREAD

**Issue by reallabs** [calimarkus/JDStatusBarNotification#76]
Hey there,

Thanks for the great library - really enjoying it. You're one of the only libraries that supports progress indicators really seamlessly and I love it.

I'm running into an issue where sometimes the status bar **never gets dismissed**, despite calling the `dismissAfter` function. Code below:

```
dispatch_async(dispatch_get_main_queue(), ^{
                    [JDStatusBarNotification showWithStatus:@"Failed to Follow User" dismissAfter:2.0 styleName:JDStatusBarStyleError];
                });
```

Any idea what might be causing this? I am only seeing it very sporadically unfortunately so I can't get a great repo. I am calling it from within a NSObject singleton that should never be deallocated. 

FYI, I also modified your repo a little bit to increase the height of the notification (made it twice as large). I only changed one line (#485 of JDStatusBarNotification.m) to the following:

```
  CGFloat height = MIN(rect.size.width, rect.size.height * 2);
```

Based on reading the dismiss function, I don't think this should be causing the issue.

Thoughts on where I can look to debug?

Thanks.

**Comment by noahls** [comment]
I had met this problem too, has any solution? 
do not change anything but use [JDStatusBarNotification showWithStatus:@"XXX" dismissAfter:5 styleName:JDStatusBarStyleWarning];

**Comment by RiverYangZiJiang** [comment]
This happend when you call [JDStatusBarNotification showWithStatus... method in an new thread, after the thread finshed, the NSTimer in the thread will never be executed, and the dismiss method of JDStatusBarNotification will never be executed, and the status bar will never dismiss. To solve this problem, you can call  
[JDStatusBarNotification showWithStatus... method in the main thread.

**Comment by reallabs** [comment]
@RiverYangZiJiang, I am dispatching the [JDStatusBarNotification showWithStatus...] to be called on the main thread though.

**Comment by calimarkus** [comment]
I just released a major update (Version 2.0). If this still happens / reproduces, please reopen this issue.
